Section 2/2.15_17_Working_with_Strings_Part1_&_Part2.rb

```diff
@@ -0,0 +1,76 @@
+# WORKING WITH STRINGS Part 1
+# String concatenation
+first_name = "hanyi"
+last_name = "koh"
+puts first_name + " " + last_name
+full_name = first_name + last_name
+
+# String interpolation
+puts "My first name is #{first_name} and my last name is #{last_name}"
+
+# Using method
+puts full_name.class    
+# => String
+p 10.methods # find all the invokable methods for the object by default
+# => [:bit_length, :digits, :|, :gcd, :lcm, ...]
+p 10.class
+# => String
```

Comment:
????

Reply:
In ruby, everything is object, including the numbers / integers themselves. In the code, the 10 is actually an object of the Integer class and we can use the methods belonging to the class. To know the invokable methods, we can use `.methods `function. `.methods` function returns a list of the names of public and protected methods of obj. This will include all the methods accessible in obj’s ancestors. To see the object is belonging to which class, we can call `.class `function and `.class` function returns the type of the object. 10 is a

Reply:
*sorry that .class function should return Integer, my typo in the comments

In ruby, everything is object, including the numbers / integers themselves. In the code, the 10 is actually an object of the Integer class and we can use the methods belonging to the class. To know the invokable methods, we can use `.methods `function. `.methods` function returns a list of the names of public and protected methods of obj. This will include all the methods accessible in obj’s ancestors. To see the object is belonging to which class, we can call `.class `function and `.class` function returns the type of the object. 10 is a Integer Class object, therefore will return Integer